### PR TITLE
Varnish clear at end of non-hotfix deployments

### PR DESCRIPTION
**Pull request by weitzman**
Hotfix deployments continue to use selective purging, not full

**Jira:** (Skip unless you are MA staff)
DP-34940

**To Test:**
- [ ] Deploy this branch to feature4 and verify  that Varnish was fully purged (search for "Varnish fully purged"). `drush ma:release --varnish --ci-branch=feature/DP-34940_varnish-clear-deployments feature4 feature/DP-34940_varnish-clear-deployments -vvv`

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by tom-fleming** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Add a changelog YAML file to this PR</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against aeec7f4f1fb0d14a8960a9b3ae0ca53831292f73
</p>

File: drush/Commands/DeployCommands.php [hunk 1]
```diff
@@ -490,6 +501,7 @@ public function purgeSelective(SiteAlias|bool $targetRecord) {
       $process = Drush::drush($targetRecord, 'ev', ["\Drupal::service('manual_purger')->purgePath('$path');"], ['verbose' => TRUE]);
       $process->mustRun();
     }
+    $this->logger()->success("Selective Purge enqueued at $target.");
```
Comment (tom-fleming) [inline drush/Commands/DeployCommands.php]:
```suggestion
    $this->logger()->success("Selective Purge enqueued at $targetRecord.");
```

File: drush/Commands/DeployCommands.php [hunk 2]
```diff
@@ -378,10 +386,13 @@ public function deploy(string $target, string $git_ref, array $options = []) {
...
-    // Process purge queue.
-    $process = Drush::drush($targetRecord, 'p:queue-work', [], ['finish' => TRUE, 'verbose' => TRUE]);
-    $process->mustRun();
-    $this->logger()->success("Purge queue worker complete at $target.");
+
+    if (!$options['varnish']) {
+      // Process purge queue.
+      $process = Drush::drush($targetRecord, 'p:queue-work', [], ['finish' => TRUE, 'verbose' => TRUE]);
+      $process->mustRun();
+      $this->logger()->success("Purge queue worker complete at $target.");
```
Comment (tom-fleming) [inline drush/Commands/DeployCommands.php]:
```suggestion
      $this->logger()->success("Purge queue worker complete at $targetRecord.");
```

Comment (weitzman) [reply drush/Commands/DeployCommands.php]: Hmmm. $targetRecord is supposed to be a SiteAlias object, and $target is a string. Are you seeing differently?

Comment (tom-fleming) [reply drush/Commands/DeployCommands.php]: Ah okay. This is likely the wrong fix then. What I was seeing was that the target was blank in the log output.

Comment (weitzman) [reply drush/Commands/DeployCommands.php]: Fix submitted

Comment (tom-fleming) [reply drush/Commands/DeployCommands.php]: Thanks! Looks good to me

**Pull request merged**